pkg/handlers/cors.go

```diff
@@ -5,8 +5,10 @@ import (
 )
 
 func CORS(w http.ResponseWriter, r *http.Request) {
+	origin := r.Header.Get("Origin")
```

Comment:
If I'm reading this correctly, we are setting origin based on the HTTP GET. This means we are trusting the client is sending an accurate header. Its easy to manipulate the headers in a HTTP request. AKA, we are trusting what is essentially user defined input. Are the headers getting validated in some other area?

Reply:
Hi @St0rmz1 The origin header will be present on any cross-origin requests, not just GET requests. The [origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) is set by the browser before sending the request to the server, it is a [forbidden header](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name) which means that the client has no way of setting or modifying the value.

Reply:
@garcialuis @St0rmz1 , to better understand this myself.  It seems like we previously had `Access-Control-Allow-Origin: *` since we cannot know the value of the origin header because KOTS may be running behind any arbitrary enterprise domain?  As I see it, this essentially produces the same behavior as the wildcard, but without the wildcard since the browser would not allow sending the session cookie along with `Access-Control-Allow-Origin: *`.

Reply:
Yes that's correct, this achieves the same behavior. Without this the browser would not allow the request to go through since it is not allowed to use the `Access-Control-Allow-Credentials: true` header while `Access-Control-Allow-Origin is *`